cv_next/app/feed/components/feed.tsx

```diff
@@ -1,82 +1,130 @@
+import { useCallback, useContext, useEffect, useRef, useState } from "react"
+import { fetchCvsForFeed } from "@/app/actions/cvs/fetchCvs"
+import CVItemRSC from "./CVItemRSC"
+import { CvsContext, CvsDispatchContext } from "@/providers/cvsProvider"
+import { ReloadButton } from "@/components/ui/reloadButton"
+import Definitions from "@/lib/definitions"
+import { useInView } from "react-intersection-observer"
+import { FilterPanel, filterObj } from "@/app/feed/components/filterPanel"
 
-export default function Feed({initialBatch} : {initialBatch: ClientCvModel[] | null}) {
-  
-  const cvsContextConsumer = useContext(CvsContext);
-  const cvsDispatchContextConsumer = useContext(CvsDispatchContext);
-  const [cvs, setCvs] = useState<ClientCvModel[] | null>(cvsContextConsumer.cvs?.length ? cvsContextConsumer.cvs : initialBatch);
-  const cvsRef = useRef<ClientCvModel[] | null>();
-  const lastCvRef = useRef<string | undefined>(cvs?.[cvs?.length]?.id);
-  const [loadMore, setLoadMore] = useState(true);
+export default function Feed() {
+  const cvsContextConsumer = useContext(CvsContext)
+  const cvsDispatchContextConsumer = useContext(CvsDispatchContext)
+  const initialCvs = cvsContextConsumer.cvs?.length
+    ? cvsContextConsumer.cvs
+    : []
+
+  const [cvs, setCvs] = useState<CvModel[]>(initialCvs)
+  const cvsRef = useRef<CvModel[] | null>()
+  const page = useRef<number>(
+    cvsContextConsumer.cvs?.length
+      ? cvsContextConsumer.page
+      : Definitions.PAGINATION_INIT_PAGE_NUMBER
+  )
+  const [loadMore, setLoadMore] = useState(true)
+  const [filters, setFilters] = useState<filterObj>({
+    searchValue: "",
+    categoryId: null
+  })
+
+
+  /**
+   * Trigger pagination when this element comes into view.
+   *
+   * @param {Function} callbackTrigger - A function that triggers pagination when called
+   * @return {JSX.Element} A div element with a ref for triggering pagination
+   */
+  function TriggerPagination({
+    callbackTrigger,
+  }: {
+    callbackTrigger: () => Promise<void>
+  }) {
+    const [triggerRef, inView] = useInView()
+
+    useEffect(() => {
+      if (inView) {
+        callbackTrigger()
+      }
+    }, [callbackTrigger, inView])
+
+    return <div ref={triggerRef}></div>
+  }
 
   useEffect(() => {
-    //update cv-related refs
-    cvsRef.current = cvs;
-    lastCvRef.current = cvs?.[cvs?.length - 1]?.id;
-  }, [ cvs ]);
-  
+    cvsRef.current = cvs
+  }, [cvs])
+
   const fetchCvsCallback = useCallback(async () => {
-    
-    //fetch cvs and handle loading states
     if (loadMore) {
-      const fetchedCvs = await fetchCvs({lastId: lastCvRef.current});
-      
-      if (fetchedCvs && fetchedCvs?.length > 0) {
-        if (fetchedCvs[fetchedCvs.length - 1]?.id === lastCvRef.current) {
-          setLoadMore(false);
-        } else {
-          setCvs((prevCvs) => [ ...(prevCvs?.length ? prevCvs : []), ...fetchedCvs ]);
-        }
+      const nextPage = page.current + 1
```

Comment:
Run prettier. @Aradon267 do we have pre-commit hook for prettier/es-lint?

Reply:
I'll check
BTW ran it but I didn't see these lines

Reply:
According to what I see we don't have it
Wanted to add it but seems that for prettier this is [deprecated](https://github.com/pre-commit/mirrors-prettier#archived). will add for eslint

Reply:
Check this: https://prettier.io/docs/en/precommit.html (especially first option)